overlord/state/state.go

```diff
@@ -482,6 +491,58 @@ func (s *State) GetMaybeTimings(timings interface{}) error {
 	return nil
 }
 
+// AddTaskStatusChangedHandler adds a callback function that will be invoked
+// whenever tasks change status.
+// NOTE: Callbacks registered this way may be invoked in the context
+// of the taskrunner, so the callbacks should be as simple as possible, and return
+// as quickly as possible, and should avoid the use of i/o code or blocking, as this
+// will stop the entire task system.
+func (s *State) AddTaskStatusChangedHandler(f func(t *Task, old, new Status)) (id int) {
+	s.reading()
```

Comment:
Maybe (a big maybe) a short comemnt explaining that "reading()" here means that we ensure the access is serialized, I was initially puzzled because we are "writing" to the internal data structures but of course "writing()" just means external state writing.

Reply:
Thanks for the review! added a comment